### PR TITLE
Refactor/Migrated Redux to TS

### PR DESCRIPTION
**Pull request by ignaciosantise**
## Summary
* Updated redux and toolkit libs
* Migrated redux slices to TS
* Migrated some constants to TS

## Github Issue
https://github.com/Psychedelic/plug-mobile/issues/497

### CODE REVIEW

File: src/redux/slices/user.ts [hunk 1]
```diff
@@ -21,165 +33,177 @@ import {
...
+      onEnd?: () => void;
+    },
+    { rejectWithValue, dispatch }
+  ) => {
     try {
       const { to, nft, icpPrice, onEnd } = params;
       const instance = KeyRing.getInstance();
       const response = await instance?.transferNFT({
         to,
+        // @ts-ignore
```
Comment (mattgle) [inline src/redux/slices/user.ts:206]: is this ok?

File: src/redux/slices/user.ts [hunk 2]
```diff
@@ -201,66 +225,80 @@ export const transferNFT = createAsyncThunk(
...
+>(
   'user/addContact',
-  async ({ contact, onFinish }, { getState, dispatch, rejectWithValue }) => {
+  async ({ contact, onFinish }, { getState, rejectWithValue }) => {
     try {
       const state = getState();
       const instance = KeyRing.getInstance();
       const res = await instance?.addContact({
-        contact: formatContactForController(contact),
+        contact: formatContactForController(contact) as Address,
```
Comment (mattgle) [inline src/redux/slices/user.ts:255]: Is it  possible to add the as Address inside the formatContactForController?

File: src/redux/slices/user.ts [hunk 3]
```diff
@@ -269,48 +307,52 @@ export const editContact = createAsyncThunk(
       });
 
       const addContactRes = await instance?.addContact({
-        contact: formatContactForController(newContact),
+        contact: formatContactForController(newContact) as Address,
```
Comment (mattgle) [inline src/redux/slices/user.ts:310]: same

File: src/redux/slices/user.ts [hunk 4]
```diff
@@ -365,20 +412,29 @@ export const getTokenInfo = createAsyncThunk(
...
 export const addConnectedApp = createAsyncThunk(
   'user/addConnectedApp',
-  /**  @param {any} [app] */
-  async (app, { getState }) => {
-    const currentConnectedApps = getState().user.connectedApps;
-    const { name, canisterList, lastConection, account } = app;
+  async (
+    app: {
+      name: string;
+      canisterList: any;
```
Comment (mattgle) [inline src/redux/slices/user.ts:428]:
```suggestion
      canisterList: WCWhiteListItem[];
```

**Review by mattgle — APPROVED** [review]
Looking good! This is huge, great work, just a few comments but i think this is approved tho

File: src/redux/slices/user.ts (hunk 1 above)
Comment (ignaciosantise) [reply src/redux/slices/user.ts:206]: this is the problem, i dont know what can i do 😆 
Screen Shot 2022-10-13 at 16 07 24

File: src/redux/slices/user.ts (hunk 2 above)
Comment (ignaciosantise) [reply src/redux/slices/user.ts:255]: `formatContactForController` is in a JS utils file that is a little bit complicated

**Review by aleacevedo — APPROVED** [review]

**Pull request merged**